Plugins/BranchTree/BranchTreeForm.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        TreeNode _tagTreeNode, _localTreeNode, _remoteTreeNode;
+        private readonly IGitUICommands _gitUiCommands;
+               
+
+        public BranchTreeForm()
+        {
+            InitializeComponent();
+            Translate();
+        }
+
+        public BranchTreeForm(GitUIBaseEventArgs gitUiCommands) : this()
+        {
+            _gitUiCommands = gitUiCommands.GitUICommands;
+            var gitModule = gitUiCommands != null ? gitUiCommands.GitModule : null;
+            
+            BranchesTreeView.BeginUpdate();
+            BranchesTreeView.Nodes.Clear();
+
+            _localTreeNode = BranchesTreeView.Nodes.Add("Local");
+            _remoteTreeNode = BranchesTreeView.Nodes.Add("Remote");
+            _tagTreeNode = BranchesTreeView.Nodes.Add("Tag");
+
+            InitLocalBranches(_localTreeNode, gitModule);
+            InitRemoteBranches(_remoteTreeNode, gitModule);
+            InitTag(_tagTreeNode, gitModule);
+          
+            BranchesTreeView.EndUpdate();
+            BranchesTreeView.NodeMouseDoubleClick += Branches_NodeMouseDoubleClick;
+        }
+
+        private void InitTag(TreeNode root, IGitModule _gitModule)
+        {
+            var tags = _gitModule.GetRefs(true, false)
+                .OrderBy(r => r.Name);
+
+            foreach (var tag in tags)
+            {
+                AddOrUpdateNode(root.Nodes, tag.Name.Split('/'), tag.CompleteName);
+            }
+        }
+
+        private void InitRemoteBranches(TreeNode root, IGitModule gitModule)
+        {
+            var branches = gitModule.GetRefs()
+                   .Where(branch => branch.IsRemote && !branch.IsTag)
+                   .OrderBy(r => r.Name);                   
+
+            var remotes = gitModule.GetRemotes(allowEmpty: true);
+            
+            foreach (var branch in branches)
+            {
+                var remote = branch.Name.Split('/').First();
+
+                if (!remotes.Contains(remote))
+                {
+                    continue;
+                }
+
+                AddOrUpdateNode(root.Nodes, branch.Name.Split('/'), branch.CompleteName);
+            }
+        }
+
+        private void InitLocalBranches(TreeNode root, IGitModule gitModule)
+        {
+            var localBranches = gitModule.RunGitCmd("branch")
+                .Split('\n')
+                .Where(branch => !string.IsNullOrWhiteSpace(branch))// first is ""
+                .OrderByDescending(branch => branch.Contains('*'))// * for current branch
+                .ThenBy(r => r)
+                .Select(line => line.Trim());// trim justify space
+            foreach (var localBranch in localBranches)
+            {
+                AddOrUpdateNode(root.Nodes, localBranch.Split('/'), localBranch);
+            }
+            
+        }
+                
+
+        private void Branches_NodeMouseDoubleClick(object sender, TreeNodeMouseClickEventArgs e)
+        {
+            if (e.Node.Tag != null)
```

Comment:
Something should be done here. Messages like this should not appear in a released product, strings should be translatable.

Reply:
How do you think what solution should be better:
1) remove branch switching from plugin at all. In this case plugin will just display branches hierarchy
2) extend plugin interface and allow plugin to switch branches. In this case plugin can swich branches

Reply:
* Use `TranslateString` for localisation. Please have a look how it is done in other plugins.
* No "_coming soon_" messages - the functionality is either there or not. Given that we do 1-2 releases a year there is no "soon".

Reply:
Not sure what is best. Of course, #3038 is complete and maybe where features should be added and keep this a viewer right now.
I suggest to copy the branch name for now.

I believe this can be merged, but RussKie takes the decision